### PR TITLE
Bigtable: Instance creation cleanup.

### PR DESCRIPTION
**Pull request by sduskis**
A Cloud Bigtable instance is a collection of clusters.  Each cluster has a node count and a location.  The python Cloud Bigtable client needs a bit of refactoring to better represent the distinction between instances and clusters.

Long term, Instance.create() should take in a complete set of variables required for creating an [instance](https://github.com/googleapis/googleapis/blob/c89b6330b6386298f8ea65e47e0b77b28294e3e7/google/bigtable/admin/v2/instance.proto#L34) and a collection of [cluster](https://github.com/googleapis/googleapis/blob/c89b6330b6386298f8ea65e47e0b77b28294e3e7/google/bigtable/admin/v2/instance.proto#L104)s.

This is PR is related to issue #5454

### CODE REVIEW

File: bigtable/tests/unit/test_instance.py [hunk 1]
```diff
@@ -330,7 +325,7 @@ def test_create_w_explicit_serve_nodes(self):
         client._instance_admin_client.bigtable_instance_admin_stub = stub
 
         # Perform the method and check the result.
-        result = instance.create()
+        result = instance.create(location_id=self.LOCATION_ID)
```
Comment (tseaver) [inline bigtable/tests/unit/test_instance.py]: Hmm, the name of the testcase suggests that we should be passing all the non-default arguments to `Instance.create` here (and asserting they are propagated correctly).

Comment (sduskis) [reply bigtable/tests/unit/test_instance.py]: I definitely agree that this needs to be changed.  I added [this comment](https://github.com/GoogleCloudPlatform/google-cloud-python/issues/5454#issuecomment-400371391) to issue #5454 to track this. This PR was intended just to be an initial refactoring, and I hope that it's OK to keep this PR as such.

There's quite a bit more work required to complete issue #5454, so is it ok to improve the test cases as a follow up PR?

**Comment by sduskis** [comment]
I merged from master, so the unit test is now fully testing the expected behavior..

**Comment by tseaver** [comment]
@sduskis 

> I merged from master, so the unit test is now fully testing the expected behavior..

I will merge when CI is green.

**Pull request merged**